staging/src/k8s.io/code-generator/cmd/validation-gen/main.go

```diff
@@ -82,3 +98,59 @@ func (args *Args) Validate() error {
 
 	return nil
 }
+
+var friendlyContextNames = map[validators.TagContext]string{
+	validators.TagContextType:  "type definitions",
+	validators.TagContextField: "field definitions (including map keys and map/slice values)",
+}
+
+func printDocs() {
+	// We need a fake context to init the validator plugins.
+	c := &generator.Context{
+		Namers:    namer.NameSystems{},
+		Universe:  types.Universe{},
+		FileTypes: map[string]generator.FileType{},
+	}
+
+	// This gets a composite validator which aggregates the many plugins.
+	validator := validators.NewValidator(c, nil, nil)
+
+	docs := builtinTagDocs()
+	docs = append(docs, validator.Docs()...)
+	slices.SortFunc(docs, func(a, b validators.TagDoc) int {
+		return cmp.Compare(a.Tag, b.Tag)
+	})
+
+	for _, doc := range docs {
+		fmt.Printf("Tag: +%s\n", clean(doc.Tag))
+		fmt.Printf("Description: %s\n", clean(doc.Description))
+		fmt.Printf("Attach to:\n")
```

Comment:
This is awesome.  Wondering out loud:  Should we output to a structured format to make it easy to produce formatted content in downstream tooling?

Reply:
I wondered that too - it's not fantasically readable as it is.  Doing exactly this data in JSON results in things like:

```
    {
        "Tag": "unionMember",
        "Description": "Indicates that this field is a member of a union.",
        "Contexts": [
            "Field"
        ],
        "Payloads": [
            {
                "Description": "\u003cjson-object\u003e",
                "Docs": "\nSchema:\n  \"union\": \u003cstring\u003e       # optional: the name of the union, if more than one exists\n  \"memberName\": \u003cstring\u003e  # optional: the discriminator value for this member, default is the field name\n"
            }
        ]
    },
```

We could change `<none>` to `{none}` and avoid some of the hex escapes.  We could avoid the embedded quotes by convention.  We could make payload docs be a []string, but now it's feeling kind of dirty.  Could make it a one-of(string, struct Schema)?